packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/link_type/read.rs

```diff
@@ -0,0 +1,47 @@
+use async_trait::async_trait;
+use error_stack::Result;
+
+use crate::{
+    ontology::types::LinkType,
+    store::{
+        postgres::resolve::{PostgresContext, Record},
+        query::{Literal, PathSegment, Resolve, ResolveError, UNIMPLEMENTED_LITERAL_OBJECT},
+    },
+};
+
+#[async_trait]
+impl<C> Resolve<C> for Record<LinkType>
+where
+    C: PostgresContext + Sync,
+{
+    async fn resolve(&self, path: &[PathSegment], context: &C) -> Result<Literal, ResolveError> {
+        match path {
+            [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
```

Comment:
I know this was approved in #1016, but I wonder if we want to return a error report here instead? If we do end up with some sort of slow-fail, it would be fine if this doesn't panic. 
No need to change it now, but perhaps we should consider this when returning all errors

Reply:
I'd like to distinguish between two different kinds of errors here: An invalid expression (e.g. comparing a string with an integer) and a maybe valid expression, which is currently not implemented. In this case I expect to resolve a `LinkType` object, so this might be a valid query. 
For clarification: For the user (i.e. REST endpoint users) both errors will result in an internal server error currently. Only the logging output differs, but neither will fail the application.